sycl/doc/CompilerAndRuntimeDesign.md

```diff
@@ -39,7 +56,16 @@ back-end. Today middle-end transformations include just a couple of passes:
   handle transformed LLVM IR.
   - Optionally: LLVM IR → SPIR-V translator.
 - **Back-end** - produces native "device" code in ahead-of-time compilation
-mode.
+mode. Shown as "Target-specific LLVM compiler". Backend is invoked at compiler
+time only in ahead-of-time compilation scenario. Otherwise it is invoked at
+runtime - JIT compilation sceanrio.
```

Comment:
```suggestion
mode. Shown as "Target-specific LLVM compiler". Backend is invoked at
compile-time only in ahead-of-time compilation scenario. Otherwise, it is
invoked at runtime (just-in-time compilation scenario).
```

Reply:
Updated per @pvchupin 's comment